### PR TITLE
Support for kitchen testing on Github Actions

### PR DESCRIPTION
**Pull request by j-rivero**
The PR implements the testing integration using Github Actions for a matrix composed of jenkins, repo, agent + chef15, chef16 using the kitchen-dokken project to launch it in docker containers. To support custom configurations for cloud testing  I've used the `KITCHEN_LOCAL_YAML` variable in combination with two kitchen.yml files. It is suboptimal to maintain two files but I did not find an easy way of not hardcoding the chef version inside these files without using an external template system.

There was a problem with jcasc and the email-ext extension, I've fixed it in  d0a0b9c . I've also documented how to inspect problems locally in 1c3624c.

### CODE REVIEW

**Comment by j-rivero** [comment]
Some updates:
 * I found the way of not having to maintain two files for different chef versions. 74f5a26 
 * I needed to remove the analysis-core plugin and the casc configuration associated 994bccb . This probably is better in a different PR
 * Added the yamlinter and fix a couple of issues 2bc4d34

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ name: ci
 "on":
   pull_request:
   push:
+    - 'master'
```
Comment (nuclearsandwich) [inline .github/workflows/main.yml]: There is no master branch. 
```suggestion
    - 'latest'
```

**Comment by nuclearsandwich** [comment]
> There was a problem with jcasc and the email-ext extension, I've fixed it in [d0a0b9c](https://github.com/ros-infrastructure/cookbook-ros-buildfarm/commit/d0a0b9ced1295dcde1f83b1af767fad531df28f0) .

This is resolved on latest with #12 

Overall this is beyond awesome! I'd like to see the removal of analysis-core moved out to a separate PR and rebased out of branch along with d0a0b9ced1295dcde1f83b1af767fad531df28f0.

**Comment by j-rivero** [comment]
> This is resolved on latest with #12
> 
> Overall this is beyond awesome! I'd like to see the removal of analysis-core moved out to a separate PR and rebased out of branch along with [d0a0b9c](https://github.com/ros-infrastructure/cookbook-ros-buildfarm/commit/d0a0b9ced1295dcde1f83b1af767fad531df28f0).

Follow in #22 and #23

**Pull request closed without merging**